sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/TableFileCatalog.scala

```diff
@@ -42,24 +43,21 @@ class TableFileCatalog(
 
   protected val hadoopConf = sparkSession.sessionState.newHadoopConf
 
+  private val fileStatusCache = FileStatusCache.getOrInitializeShared(new Object(), sparkSession)
```

Comment:
The way we currently refresh tables is by dropping the reference to TableFileCatalog and letting it get GC'ed. Given this strategy, making it private is the simplest way to ensure refresh actually works correctly -- otherwise you have to carefully test that refresh also invalidates shared cache entries.